modules/javafx.web/src/main/native/Source/WebCore/platform/java/FileSystemJava.cpp

```diff
@@ -232,21 +232,76 @@ String openTemporaryFile(const String&, PlatformFileHandle& handle)
     return String();
 }
 
-PlatformFileHandle openFile(const String&, FileOpenMode)
+PlatformFileHandle openFile(const String& path, FileOpenMode mode)
 {
-    notImplemented();
-    return invalidPlatformFileHandle;
+    JNIEnv* env = WebCore_GetJavaEnv();
+    static jmethodID mid = env->GetStaticMethodID(
+            GetFileSystemClass(env),
+            "fwkOpenFile",
+            "(Ljava/lang/String;Ljava/lang/String;)Ljava/io/RandomAccessFile;");
+    ASSERT(mid);
+
+    // fileMode initialized to NULL to avoid compilation warning
+    jstring fileMode = NULL;
+    if (mode == FileOpenMode::Read) {
```

Comment:
Since we support only FileReader as of now, it is better to return error when FileMode is Write.